src/main/kotlin/com/exactpro/th2/processor/core/event/controller/EventController.kt

```diff
@@ -44,15 +43,14 @@ internal class EventController(
         get() = super.isStateComplete && eventState.isStateEmpty
 
     override fun actual(batch: EventBatch) {
-        updateState {
-            for (event in batch.eventsList) {
-                updateState(batch, event)
+        var needSignal  = false
+        for (event in batch.eventsList) {
+            needSignal = needSignal or updateState(event)
```

Comment:
Maybe we should separate the right part from the `or` expression. I am not sure if there are guarantees that the right part is always executed and the behavior want change in the future
```suggestion
            val updateResult = updateState(event)
            needSignal = needSignal or updateResult
```